### DISCUSSION TITLE
add image/avatar preview field to the table

### DISCUSSION THREAD

**Discussion by rama-adi** [filamentphp/filament#52] [Ideas]
A new field on the table that adds image previews for image fields. Like this:
![image](https://user-images.githubusercontent.com/17980909/109699571-b169a380-7bcb-11eb-97e9-d784e0a53f9b.png)

**Comment by rama-adi** [comment]
I decided to make my own. here's how it looks like:
![image](https://user-images.githubusercontent.com/17980909/109710586-cc8ee000-7bd8-11eb-853d-95b0d50bd19b.png)
if you wanna follow along, just do the following:
1. create a new class that extends `\Filament\Tables\Columns\Column`
2. create public properties and setter that you want. i added `disk`.
```php
 public $disk = "public";

    public function disk($disk = 'public')
    {
        $this->disk= $disk;
        return $this;
    }
```
3. implement the `renderCell` method with your view
```php
 public function renderCell($record)
    {

        return view('cells.avatar-preview', [
            'column' => $this,
            'record' => $record,
            'disk'  => $this->disk
        ]);
    }
```
5. Customize the view
```blade
<img class="h-8 rounded-full" src="{{Storage::disk($disk)->url($column->getValue($record))}}" alt="Avatar preview">
```
6. Use on your table
```php
 return $table
            ->columns([
               AvatarPreview::make('profile_photo'),
            ])
            ->filters([
                //
            ]);
```
Feel free to modify my implementation

**Comment by ManojKiranA** [comment]
Its added to core. check out the [docs](https://filamentadmin.com/docs/1.x/admin/tables#image)
